kratos/utilities/parallel_utilities.cpp

```diff
@@ -24,8 +25,12 @@
 #include "includes/lock_object.h"
 
 
-namespace Kratos
-{
+namespace Kratos {
+
+namespace {
+    std::mutex GlobalLockInitializerMutex;
```

Comment:
this is the important part
If this works (which I think it does as it is only declared and used once here)
then we can use it for initializing the `GlobalLock` ...